phase/install_controllers.go

```diff
@@ -79,6 +79,10 @@ func (p *InstallControllers) Run() error {
 }
 
 func (p *InstallControllers) waitJoined(h *cluster.Host) error {
+	port := p.Config.Spec.K0s.Config.Dig("spec", "api", "port").(int)
```

Comment:
I think this needs something like:

```go
port := 6443
if p, ok := p.Config.Spec.K0s.Config.Dig("spec", "api", "port").(int); ok {
  port = p
}
```

Reply:
Gives `FATA[0015] PANIC: interface conversion: interface {} is nil, not int` when `spec.api.port` not set otherwise.

Reply:
Oh my bad! You are :100:% spot on.

Reply:
is this fixed in the commits leading to 83dc9b34caf7727f377865b743be9e82777597a8? if so, this is also failing with that error (which I don't known if its because this particular line that I'm commenting on):

```console
# k0sctl kubeconfig --config /vagrant/shared/k0sctl.yaml
FATA PANIC: interface conversion: interface {} is nil, not int 
```